### DISCUSSION TITLE
.mpy unsolicited print

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#10670] [ESP32]
Running a .py on a generic ESP32 cam board, it wakes up takes a pic, uploads it to a server & deepsleeps till the next wakeup. Once I'd got all the bugs out I converted it to a code.mpy file that is run from boot.py with 'import code'.

The program still runs OK in it's .mpy guise but in the middle of the progress prints is 
`E (13690) gpio: gpio_install_isr_service(449): GPIO isr service already installed`
that I didn't get when it was running as a .py  It's not a problem. the program still runs OK but it's occurrence in the middle of the program, when I've long since finished configuring the gpio, has me intrigued.

Curious why running as a .mpy is causing this change?

**Comment by davefes** [comment]
When I developed my camera app using https://github.com/shariltumin/esp32-cam-micropython-2022
I also got that message sometimes and I am just using .py files.

**Reply by Josverl** [reply]
The prefix 'e (###)' hints to an esp. osdebug message, rather than a print output.
https://docs.micropython.org/en/latest/library/esp.html#esp.osdebug

Why, possible timing related

**Reply by Josverl** [reply]
Possibly related : https://esp32.com/viewtopic.php?t=23104

**Reply by kjm1102** [reply]
Interesting reading, I love the after thought at the end, viz (I'd like to also state that in general, it's not really our policy to generate error messages like that that you can 'just ignore', even if it happens to work out here.)
